doc/v2/documentation_data_format.rst

```diff
@@ -0,0 +1,1172 @@
+       | (NOTE: ``log`` without explicit
+       | base is ``ln``, not ``log10``)
+     - float[, float]
+     - float
+   * - | ``sin``
+       | ``cos``
+       | ``tan``
+       | ``cot``
+       | ``sec``
+       | ``csc``
+     - trigonometric functions
+     - float
+     - float
+   * - | ``arcsin``
+       | ``arccos``
+       | ``arctan``
+       | ``arccot``
+       | ``arcsec``
+       | ``arccsc``
+     - inverse trigonometric functions
+     - float
+     - float
+   * - | ``sinh``
+       | ``cosh``
+       | ``tanh``
+       | ``coth``
+       | ``sech``
+       | ``csch``
+     - hyperbolic functions
+     - float
+     - float
+   * - | ``arcsinh``
+       | ``arccosh``
+       | ``arctanh``
+       | ``arccoth``
+       | ``arcsech``
+       | ``arccsch``
+     - inverse hyperbolic functions
+     - float
+     - float
+   * - | ``piecewise(``
+       |     ``true_value_1,``
+       |       ``condition_1,``
+       |     ``[true_value_2,``
+       |       ``condition_2,]``
+       |     ``[...]``
+       |     ``[true_value_n,``
+       |       ``condition_n,]``
+       |     ``otherwise``
+       | ``)``
+     - | The function value is
+       | the ``true_value*`` for the
+       | first ``true`` ``condition*``
+       | or ``otherwise`` if all
+       | conditions are ``false``.
+     - | ``*value*``: all float or all bool
+       | ``condition*``: all bool
+     - float
+   * - ``abs(x)``
+     - | absolute value
+       | ``piecewise(x, x>=0, -x)``
+     - float
+     - float
+   * - ``sign(x)``
+     - | sign of ``x``
+       | ``piecewise(1, x > 0, -1, x < 0, 0)``
+     - float
+     - float
+   * - | ``min(a, b)``
+       | ``max(a, b)``
+     - | minimum / maximum of {``a``, ``b``}
+       | ``piecewise(a, a<=b, b)``
+       | ``piecewise(a, a>=b, b)``
+     - float, float
+     - float
+
+..
+   END TABLE Supported functions
+
+
+Boolean <-> float conversion
+++++++++++++++++++++++++++++
+
+Boolean and float values are implicitly convertible. The following rules apply:
+
+bool -> float: ``true`` is converted to ``1.0``, ``false`` is converted to
+``0.0``.
+
+float -> bool: ``0.0`` is converted to ``false``, all other values are
+converted to ``true``.
+
+Operands and function arguments are implicitly converted as needed. If there is
+no signature compatible with the given types, Boolean
+values are promoted to float. If there is still no compatible signature,
+float values are demoted to boolean values. For example, in ``1 + true``,
+``true`` is promoted to ``1.0`` and the expression is interpreted as
+``1.0 + 1.0 = 2.0``, whereas in ``1 && true``, ``1`` is demoted to ``true`` and
+the expression is interpreted as ``true && true = true``.
+
+
+.. _identifiers:
+
+Identifiers
+-----------
+
+* All identifiers in PEtab may only contain upper and lower case letters,
+  digits and underscores, and must not start with a digit. In PCRE2 regex, they
+  must match ``[a-zA-Z_][a-zA-Z_\d]*``.
```

Comment:
Above, this was written like
```suggestion
  must match ``[a-zA-Z_]\w*``.
```

Reply:
Then the other occurrence needs to be fixed. I think we wanted to restrict it to a-zA-Z0-9_.

Reply:
I think it's the same (`[\w]` = `[a-zA-Z0-9_]` = `[a-zA-Z_\d]`), was just commenting that it wasn't written the same way every time.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Cheatsheet

Reply:
That depends on whether unicode support is enabled or not. I'd prefer being more explicit here.

```
$ python -c "import re; print(re.match('\w', 'Ψ'))"
<re.Match object; span=(0, 1), match='Ψ'>
```

Reply:
Agreed!